module-system/sov-app-template/src/lib.rs

```diff
@@ -202,13 +206,33 @@ where
 
         self.working_set = Some(batch_workspace);
         BatchReceipt {
-            batch_hash: [0u8; 32], // TODO: calculate the hash using Context::Hasher;
+            batch_hash: batch_data_and_hash.hash,
             tx_receipts,
             inner: SequencerOutcome::Rewarded,
         }
     }
 }
 
+struct BatchDataAndHash {
+    hash: [u8; 32],
+    data: Vec<u8>,
+}
+
+impl BatchDataAndHash {
+    fn new<C: Context>(batch: impl Buf) -> BatchDataAndHash {
+        let mut reader = batch.reader();
+        let mut batch_data = Vec::new();
+        reader
```

Comment:
The only way I found to calculate the batch hash was by converting it to a slice of bytes.